docs/applications/nublado/major-upgrade.rst

```diff
@@ -0,0 +1,313 @@
+those namespaces) or by executing something like ``for i in '' 2; do
+kubectl get ns | grep nublado${i}- | awk '{print $1}' | xargs kubectl
+delete ns; done``.
+
+Bring up applications
+"""""""""""""""""""""
+#. Synchronize the ``nublado2`` application.  This will move the
+   default spawner to the ``/n2`` route.
+#. Test that you can get a lab and that that lab allows use of the
+   command line and cell executions with the new spawner at ``/n2``.  If
+   you find errors, troubleshoot them.  If nublado v2 was working
+   correctly beforehand, there should be no errors.
+#. Synchronize the ``nublado`` application.  This will turn on the
+   nublado v3 spawner on the default route ``/nb``.  At this point users
+   will start returning (whether or not you call an all-clear, which you
+   should not yet; users don't pay attention to such things).
+#. Re-enable mobu, if you disabled it.  Wait to make sure it's correctly
+   running notebooks, or test a few yourself.
+
+Remove nublado v2
+-----------------
+
+Pre-deployment Preparation
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+As before, these steps can be done at any time, and should be done in
+advance of the maintenance window.
+
+Communicate to user base
+""""""""""""""""""""""""
+
+Let your users know that ``/n2`` will no longer available.  Announce that
+that maintenance window will be disruptive and will destroy any user
+sessions spawned via ``/n2``.  Hopefully that will be none or at least
+very few.
+
+Turn off nublado v2
+"""""""""""""""""""
+
+Delete the entries ``cachemachine: true``, ``moneypenny: true``, and
+``nublado2: true`` from your instance-specific values file in
+``environment``.
+
+Deployment
+^^^^^^^^^^
+
+During the deployment window:
+
+#. Synchronize the ``science-platform`` application.  This will delete
+   the ``nublado2``, ``cachemachine``, and ``moneypenny`` applications.
+#. Remove any user sessions spawned by ``nublado2``.  This is harder to
+   do from the UI: you will need to find any users in ``nublado-users``
+   with a namespace called ``nublado2-<username>`` and remove only those
+   namespaces.  From the CLI, it's easier: ``kubectl get ns | grep
+   nublado2- | awk '{print $1}' | xargs kubectl delete ns``.
+#. Delete the values files for your installation from Phalanx's
+   ``applications/nublado2``, ``applications/cachemachine``, and
+   ``applications/moneypenny``.
+#. If you're feeling ambitious, you can remove the ``jupyterhub``
+   database and ``jovyan`` user from your Postgres instance (or ask your
+   DBA to do so), and remove the ``nublado2``, ``cachemachine``, and
+   ``moneypenny`` entries from your instance's Vault envlave.  In
+   practice we have not been doing this, because the resources consumed
+   are minimal.
+
+When Migration Is Complete For All Sites
+-----------------------------------------
+
+Once all sites are running nublado v3, and none are still running
+nublado v2, ``applications/nublado``, ``applications/moneypenny``, and
```

Comment:
applications/nublado2?

Reply:
Yes, and you're right that we should have called this `/nublado3` but we didn't, and we're almost done with the migration.